### ISSUE TITLE
More builtins/intrinsics

### ISSUE THREAD

**Issue by Wuerfel21** [totalspectrum/spin2cpp#275]
AFAICT there aren't any intrinsic functions for the more peculiar P2 instructions (MOVBYTS, MERGEx/SPLITx, MUXNIBS/NITS, ADDPIX/MULPIX, etc).

MOVBYTS should probably also be exposed as `__builtin_bswap*`. REV should be `__builtin_bitreverse*` (note that GCC doesn't have that one, but Clang does). I also don't think we have a `__builtin_parity` (OR x,#0 wc)...

**Comment by totalspectrum** [comment]
I think most of the P2 specific features are probably best left to the user to implement in inline assembly (otherwise we have to make P1 and bytecode versions of them :( ). But I agree that __builtin_parity would be useful, and I've added that. We already had __builtin_propeller_rev and __builtin_bitrev; I've renamed the latter to __builtin_bitreverse32 for clang compatibility. __builtin_bswap32 and __builtin_bswap16 probably would be useful. Is there a fast way to do those in bytecode?
